### PR TITLE
Retry if the KoreBuild download fails

### PR DESCRIPTION
**Pull request by victorhurdugaci**
Fixes https://github.com/aspnet/KoreBuild/issues/20 https://github.com/aspnet/KoreBuild/issues/5

@pranavkm @natemcmaster @BrennanConroy

### CODE REVIEW

File: template/build.sh [hunk 1]
```diff
@@ -18,7 +18,18 @@ if test ! -d $buildFolder; then
...
+    retries=6
+    until (wget -O $localZipFile $koreBuildZip 2>/dev/null || curl -o $localZipFile --location $koreBuildZip 2>/dev/null)
+    do
+        echo "Failed to download '$koreBuildZip'"
+        if [ "$retries" -le 0 ]; then
+            exit 1
+        fi
+        retries=$((retries - 1))
+        echo "Waiting 10 seconds before retrying. Retries left: $retries"
+        sleep 1s
```
Comment (BrennanConroy) [inline template/build.sh]: 10s?

**Comment by BrennanConroy** [comment]
Looks good

**Comment by natemcmaster** [comment]
Assuming it works (and you fix to `sleep 10s`) :shipit:

File: template/build.ps1 [hunk 2]
```diff
@@ -20,7 +50,10 @@ if (!(Test-Path $buildFolder)) {
 
     $localZipFile="$tempFolder\korebuild.zip"
     
-    Invoke-WebRequest $koreBuildZip -OutFile $localZipFile
+    DownloadWithRetry -url $koreBuildZip -downloadLocation $localZipFile -retries 6
```
Comment (pranavkm) [inline template/build.ps1:53]: Isnt' this a bit much (a minute of retries?)

File: template/build.ps1 [hunk 3]
```diff
@@ -1,3 +1,33 @@
...
+        }
+        catch
+        {
+            $exceptionMessage = $_.Exception.Message
+            Write-Host "Failed to download '$url': $exceptionMessage"
+            if ($retries -gt 0) {
+                $retries--
+                Write-Host "Waiting 10 seconds before retrying. Retries left: $retries"
+                Start-Sleep -Seconds 10
+
```
Comment (pranavkm) [inline template/build.ps1:20]: Extra line.

**Pull request merged**